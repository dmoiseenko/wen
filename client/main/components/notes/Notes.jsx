import React from 'react';
import PropTypes from 'prop-types';

import Note from './Note';


export default function Notes({ notes, loading }) {
  return (
    <div className="content">
      <ul>
        {
          !loading &&
          notes.map(note =>
            (
              <Note
                key={note.id}
                note={note}
              />
            ))
        }
      </ul>
    </div>
  );
}

Notes.propTypes = {
  notes: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
};

Notes.defaultProps = {
  notes: [],
};
