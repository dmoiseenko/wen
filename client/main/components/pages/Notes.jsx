import React from 'react';

import NotesContainer from '../notes/Notes.container';
import NoteFormContainer from '../notes/NoteForm.container';


export default () => (
  <div className="section">
    <div className="container">
      <h1 className="title is-2">
        Notes
      </h1>
      <NotesContainer />
      <NoteFormContainer />
    </div>
  </div>
);
