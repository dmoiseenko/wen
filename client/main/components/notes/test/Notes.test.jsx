import React from 'react';
import { shallow } from 'enzyme';

import Notes from '../Notes';


it('should render notes when loading is false', () => {
  const props = {
    loading: false,
    notes: [{
      id: 1
    }]
  };

  const wrapper = shallow(
    <Notes {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should not render notes when loading is true', () => {
  const props = {
    loading: true,
    notes: [{
      id: 1
    }]
  };

  const wrapper = shallow(
    <Notes {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});
