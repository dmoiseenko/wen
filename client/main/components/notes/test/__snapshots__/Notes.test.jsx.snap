// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render notes when loading is true 1`] = `
<div
  className="content"
>
  <ul />
</div>
`;

exports[`should render notes when loading is false 1`] = `
<div
  className="content"
>
  <ul>
    <Note
      note={
        Object {
          "id": 1,
        }
      }
    />
  </ul>
</div>
`;
